src/ripple/app/paths/cursor/DeliverNodeReverse.cpp

```diff
@@ -37,15 +37,18 @@ namespace path {
 
 // To deliver from an order book, when computing
 TER PathCursor::deliverNodeReverse (
-    AccountID const& uOutAccountID,  // --> Output owner's account.
-    STAmount const& saOutReq,      // --> Funds requested to be
-                                   // delivered for an increment.
-    STAmount& saOutAct) const      // <-- Funds actually delivered for an
-                                   // increment.
+    AccountID const& uOutAccountID, // --> Output owner's account.
+    STAmount const& saOutReq,       // --> Funds requested to be
+                                    // delivered for an increment.
+    STAmount& saOutAct,             // <-- Funds actually delivered for an
+                                    // increment.
+    bool restartDir) const
 {
     TER resultCode   = tesSUCCESS;
 
-    node().directory.restart(multiQuality_);
+    if (restartDir)
+        for (int i = nodeIndex_; i >= 0 && !node (i).isAccount(); --i)
```

Comment:
Tiny quibble - wrap multiline statement in brances?

Question: why are we doing this backward?  I assume the simple answer is "because it's reverse" but that's only superficially convincing. 

And why do we stop when we hit the first account?

I always try to answer these by writing up a comment, but, well, I failed.  :-)  A nice comment here would be helpful for the next engineer reading this.